aws/resource_aws_s3_bucket_object.go

```diff
@@ -219,6 +225,16 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		putInput.ContentType = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("metadata"); ok {
+		meta := make(map[string]*string)
+		for key, value := range v.(map[string]interface{}) {
+			valueStr := fmt.Sprintf("%v", value)
+			meta[key] = &valueStr
+		}
+
+		putInput.Metadata = meta
```

Comment:
The above conversion logic can be removed with using the existing `stringMapToPointers` helper function:

```suggestion
		putInput.Metadata = stringMapToPointers(v.(map[string]interface{}))
```